internal/addrs/parse_ref.go

```diff
@@ -333,6 +333,25 @@ func parseRef(traversal hcl.Traversal) (*Reference, tfdiags.Diagnostics) {
 			Remaining:   remain,
 		}, diags
 
+	case "run":
```

Comment:
I don't think this logic should be a part of the general `parseRef` function, as it would leak into the regular TF operations.
From looking at the code, I see there is the [ParseRefFromTestingScope](https://github.com/opentofu/opentofu/blob/8ce1b42085b36e5a2f8a3fd536623df2386873ba/internal/addrs/parse_ref.go#L93) function., and I believe this logic should fit there.